site/content/docs/contributing/developer-docs/architecture.md

```diff
@@ -0,0 +1,57 @@
+---
+title: Architecture
+weight: 2
+---
+
+# Architecture and Plugin Startup
+
+This document describes the distributed architecture of Hipcheck and how plugins
+get started.
+
+Hipcheck is a relatively simple multiprocessed tool that follows a star topology
+(there is a single central node to which all other nodes connect). Users invoke
+the main Hipcheck binary, often referred to as "Hipcheck core" or `hc`, on the
+command line, and provide a [policy file][policy_file] which specifies the set
+of top-level plugins to use during analysis. Once Hipcheck resolves these
+plugins and their dependencies, it starts each plugin in a separate child
+process. Once all plugins are started and initialized, Hipcheck
+core enters the analysis phase. During this phase it acts as a simple hub for
+querying top-level plugins and relaying queries between plugins, as plugins are
+intended to only communicate with each other through the core.
```

Comment:
Let's add a line after this to explain that this "only through the core" rule is so the core can cache and reuse the results of queries.